### PR TITLE
[MIG] migrate business_requirement_deliverable_cost from 10.0 to 12.0

### PR DESCRIPTION
**Pull request by ruter-lyu**
@elicoidal

### CODE REVIEW

File: business_requirement_deliverable_cost/models/business.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+        self.ensure_one()
+        super(BusinessRequirementResource, self).product_id_change()
+        self._set_sales_price()
+        self._set_cost_price()
+
+    @api.multi
+    @api.onchange('uom_id', 'qty')
+    def product_uom_change(self):
+        self.ensure_one()
+        # Calculte the sales_price_unit
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/models/business.py]:
```suggestion
        # Calculate the sales_price_unit
```

File: business_requirement_deliverable_cost/models/business.py [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+        self._set_sales_price()
+        # Calculate the unit_price
+        self._set_cost_price()
+
+
+class BusinessRequirementDeliverable(models.Model):
+    _inherit = "business.requirement.deliverable"
+
+    @api.multi
+    def _default_currency(self):
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/models/business.py]: all currency at deliverable level should be transferred to BRD module.
(migration script to be considered)

File: business_requirement_deliverable_cost/readme/INTRODUCTION.rst [hunk 3]
```diff
@@ -0,0 +1,10 @@
+This module improves the cost control of the original module with the following:
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/readme/INTRODUCTION.rst:1]: BTW: there is no INTRODUCTION file in fragment.
I will rework the README later on

File: business_requirement_deliverable_cost/security/business_requirement_deliverable_security.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+© 2016-2019 Elico Corp (https://www.elico-corp.com).
+License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+-->
+<odoo>
+        <record id="group_business_requirement_estimation" model="res.groups">
+            <field name="name">Business Requirement Estimation</field>
+            <field name="implied_ids" eval="[(4, ref('business_requirement.group_business_requirement_user'))]"/>
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/security/business_requirement_deliverable_security.xml]: improve readability

**Review by elicoidal — CHANGES_REQUESTED** [review]
No functional test yet

**Review by JoJoJoJoJoJoJo — APPROVED** [review]
LGTM

**Comment by elicoidal** [comment]
@ruter-lyu we might need to review the currency fields. 

Here is my basic idea:
* Currency should be at BR level, not BRD (there might be some related fields for calculation but set up is at BR level). Only one currency per BR of course.
* Currency is linked to PL by onchange function.
* All currency fields should be moved accordingly to BRD module (where the first fields about price are added)
* in this module you will have some functions for sure based on the added fields.
* Cost fields are not impacted by currency (they are always expressed in the company currency)
* GP report should be in company currency (to be discussed whether it makes sense to have both currencies) 

We will probably later refactor all this part to move it to simple SO to follow standard Odoo logic.

**Comment by pedrobaeza** [comment]
Closing as old and not green

**Pull request closed without merging**